### ISSUE TITLE
Oauth not working

### ISSUE THREAD

**Issue by kyuzumaki** [Thomas55555/husqvarna_automower#568]
**Describe the bug**
Adding the automower integration fails at oauth. It gives an error 500 internal server error.

**Expected behavior**
Previously worked fine. Expected it to authenticate normally
 The log entry thats generated it this 

Logger: homeassistant.helpers.config_entry_oauth2_flow
Source: helpers/config_entry_oauth2_flow.py:211
First occurred: 8:11:49 (4 occurrences)
Last logged: 8:16:35

Token request failed (invalid_client): Client or client secret is invalid

Running latest home assistant image 2024.1.16 in a docker container.

**Comment by Thomas55555** [comment]
Have you checked your Client id or client secret? Maybe a trailing whitespace?

**Comment by kyuzumaki** [comment]
Yeah it all seems ok, cant see any extra spaces there. Is there any way to manually copy that info from the response url to the integration?

**Comment by dawiinci** [comment]
I get this too after upgrading to Python 3.12.
